design-docs/improved-console.md

```diff
@@ -0,0 +1,283 @@
+
+### Open issues
+* How is `--continuous` build displayed? During execution? While waiting? (as we allow processes to stay running)
+* When things fail, will the new CLI look like a regular failure or something else?
+* How does `--parallel` (or not) affect display?
+* How does having more operations than workers affect display?
+
+## Story: Display build progress as a progress bar through colorized output
+Render ProgressEvents with build completeness information as a progress bar.
+ 
+Intended to give the user a very fast way of telling whether the build will be finished soon or not.
+The work-in-progress (yellow) section shows how much of the build would be complete if all
+Operations displayed in the progressArea completed.
+
+### User-visible Changes
+We can visually represent complete, in-progress, and un-started tasks of build using colorized output:
+
+`#####green#####>##yellow##>#####black#####> 40% Building`
+
+On ANSI terminals, we can use empty spaces with different background colors for the "bars" and
+A red background shall be used for failed tasks. 
+
+### Implementation
+`DefaultGradleLauncherFactory` registers a `BuildProgressListener` listening
+to events from `BuildProgressLogger` and forwards them to a `ConsoleBackedBuildProgressRenderer`.
+ 
+`ProgressLine` consists of a `ProgressBar`, status text, and a width. It doesn't know anything about builds. 
+
+`ProgressBar` has text and width. Weakly modeling this allows for alternate `ProgressLine`s that can
+potentially be configured by users.
+
+The `DefaultProgressBar`, though, has a width and a list of `ProgressBarRegion`s,
+each of which have a width and a `RegionType` of `SUCCEEDED`, `FAILED`, `IN_PROGRESS`, or `UNSTARTED`. 
+Each`RegionType` is associated with a color. This allows us to show failures in arbitrary locations
+as needed with `--continue`
+
+### Test Coverage
+* Logs should be streamed with plain output and not throttled when not attached to a Console
+* The TAPI is unaffected except for additional BuildOperations now published
+* Logs should be streamed with plain output when attached Terminal lacks of color or cursor support
+* Renders well on default macOS Terminal, default Windows Console, Cygwin, PowerShell and common Linux Terminals
+
+### Open Issues
+* Characters outside ASCII will give a more polished look, but with risk of poor display if console font doesn't
+support them
+* ASCII-based, unstyled alternatives:
+```
+[##                        ] 6% Building
+[‡‡‡‡                      ] 12% Building
+‹===================       › 60% Building
+«==========================» 100% BUILD SUCCESS
+```
+
+## Story: Display intra-operation progress
+When we know in advance the number of things to be processed, we display progress in [Complete / All] format. 
+Provide APIs to optionally add # complete and # of all items.  
+
+For example:
```

Comment:
I think we would just have to register a Listener and then render that we reached a certain stage. I think it would give a good indication of "as a user i want to know when the actual of execution of tasks really starts", "if configuration takes longer than I expected it to take - maybe there's something wrong my code".